homeassistant/components/keenetic_ndms2/__init__.py

```diff
@@ -1 +1,44 @@
 """The keenetic_ndms2 component."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import Config, HomeAssistant
+
+from .const import DOMAIN
+from .router import KeeneticRouter
+
+
+async def async_setup(hass: HomeAssistant, _config: Config) -> bool:
+    """Set up configured entries."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up the component."""
+
+    router = KeeneticRouter(hass, config_entry)
+    await router.async_setup()
+
+    hass.data.setdefault(DOMAIN, {})[config_entry.entry_id] = router
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device_registry.async_get_or_create(
+        config_entry_id=config_entry.entry_id,
+        connections={(DOMAIN, router.host)},
```

Comment:
`router.host` is stable since the value is entered by the user and checked that this address is reachable. This won't change.

Reply:
If it's an ip address we don't allow it as unique id.

Reply:
It's a hostname of the router you use to connect to it, it might be (and most probably will be) and IP address of the router.
If it's an IP it's a static IP since your network is organized around this.
I don't see issues here since it's used for connections not for the unique id.

Reply:
As for the mac address, there is no one special. The router has a bunch of mac addresses in different networks it handles

Reply:
We don't allow ip as id, please remove this.

Reply:
OK. I'll remove that. But I don't get why it's wrong for `connections`

Reply:
Removed and changed the device ID to include config entry id instead as this device represents the configured router.